lib/Cake/Utility/Hash.php

```diff
@@ -59,7 +59,7 @@ public static function get(array $data, $path, $default = null) {
 		}
 
 		foreach ($parts as $key) {
-			if (is_array($data) && isset($data[$key])) {
+			if ( is_array($data) && array_key_exists($key,$data) ) {
```

Comment:
That is one of the reasons we have favoured `isset()` over `array_key_exists()`. Also keys existing but containing null wasn't thought to be that important of a use case back then.